corehq/apps/app_manager/tests/test_suite_multi_select_case_list.py

```diff
@@ -407,3 +415,162 @@ def test_select_parent_that_selects_other_different_case_type(self):
             ('datum', 'case_id'),
         ])
         self.assert_form_datums(self.m1f0, 'case_id')
+
+
+@patch('corehq.apps.app_manager.helpers.validators.domain_has_privilege', return_value=True)
+@patch('corehq.apps.builds.models.BuildSpec.supports_j2me', return_value=False)
+@patch('corehq.util.view_utils.get_url_base', new=lambda: "https://www.example.com")
+@patch.object(Application, 'enable_practice_users', return_value=False)
+@patch_validate_xform()
+@patch_get_xform_resource_overrides()
+@flag_enabled('USH_CASE_LIST_MULTI_SELECT')
+class MultiSelectEndOfFormNavTests(SimpleTestCase, TestXmlMixin):
+    CASE_TYPE = 'noun'
+
+    def setUp(self):
+        # All of these tests use the same app structure, which has the following menus:
+        # m0 is a standalone single-select menu
+        # m1 is a standalone multi-select menu
+        # m2/m3 are a parent single select and child multi-select
+        # m3/m4 are a parent multi-select and child single select
+        # All menus use the same case type and all forms require a case.
+        self.factory = AppFactory(domain="multiple-referrals-eof-nav-test", build_version='2.43.0',
+                                  include_xmlns=True)
+
+        self.single_loner, form0 = self.factory.new_basic_module('Single Loner', self.CASE_TYPE)
+        self.multi_loner, form1 = self.factory.new_basic_module('Multi Loner', self.CASE_TYPE)
+        self.multi_loner.case_details.short.multi_select = True
+
+        self.single_parent, form2 = self.factory.new_basic_module('Single Parent', self.CASE_TYPE)
+        self.multi_child, form3 = self.factory.new_basic_module('Multi child', self.CASE_TYPE,
+                                                                parent_module=self.single_parent)
+        self.multi_child.case_details.short.multi_select = True
+
+        self.multi_parent, form4 = self.factory.new_basic_module('Multi Parent', self.CASE_TYPE)
+        self.multi_parent.case_details.short.multi_select = True
+        self.single_child, form5 = self.factory.new_basic_module('Single Child', self.CASE_TYPE,
+                                                                 parent_module=self.multi_parent)
+
+        for module in self.factory.app.get_modules():
+            for form in module.get_forms():
+                form.requires = 'case'
+
+    @patch('corehq.apps.app_manager.helpers.validators.domain_has_privilege', return_value=True)
+    def test_block_form_linking(self, *args):
```

Comment:
There is a little bit of form linking that works - e.g., linking to a survey form - but I don't think it's worth the extra complexity of picking those out unless a project has a use case for them.